### ISSUE TITLE
on_motion_detected: Motion does not appear to be running a bash script

### ISSUE THREAD

**Issue by CaptainMayotoast** [Motion-Project/motion#1108]
0. Reviewed guide and contributing documents? (Yes/No): Yes
1. version [x.y.z, hash, other]: 4.2.2
2. installed as a package or compiled from sources [deb, rpm, git, other]: deb
3. standalone or part of third party [motion, MotionEyeOS, other]: motion
4. video stream source [V4L (card or USB), net cam (mjpeg, rtsp, other), mmal]: USB Logitech web camera
5. hardware [x86, ARM, other]: x86
6. operating system [16.04, Stretch, etc, FreeBSD, other]: Ubuntu 18.04

**The Problem**

I appear to be unable to launch a script (please see 'The Expected Result' section below) with Motion when motion activity is seen.

**The Expected Result**

My intention is to copy a file from the home folder for my user to a new file within the same folder to test Motion's script triggering capability for some other use cases.

My bash script is simple: 

> #!/bin/sh
> cp /home/***removed***/test_file1 /home/***removed***/test_file2
(I've also tried #!/bin/bash)

**The Actual Result**

The file is not copied, unfortunately.

**Motion Logs**

The initial setup logs for Motion as a systemd service are:

> Feb 07 23:39:39 ***removed***motion[11319]: [0:motion] [NTC] [STR] webu_start_ctrl: Started webcontrol on port 8080
> Feb 07 23:39:39 ***removed***motion[11319]: [0:motion] [NTC] [ENC] ffmpeg_global_init: ffmpeg libavcodec version 57.107.100 libavformat version 57.83.100
> Feb 07 23:39:39 ***removed***motion[11319]: [0:motion] [NTC] [ALL] translate_init: Language: English
> Feb 07 23:39:39 ***removed***motion[11319]: [0:motion] [NTC] [ALL] motion_start_thread: Camera ID: 0 is from /etc/motion/motion.conf
> Feb 07 23:39:39 ***removed***motion[11319]: [0:motion] [NTC] [ALL] motion_start_thread: Camera ID: 0 Camera Name: (null) Device: /dev/video0
> Feb 07 23:39:39 ***removed***motion[11319]: [0:motion] [NTC] [ALL] main: Waiting for threads to finish, pid: 11319
> Feb 07 23:39:39 ***removed***motion[11319]: [1:ml1] [NTC] [ALL] motion_init: Camera 0 started: motion detection Enabled
> Feb 07 23:39:39 ***removed***motion[11319]: [1:ml1] [NTC] [VID] vid_start: Opening V4L2 device
> Feb 07 23:39:39 ***removed***motion[11319]: [1:ml1] [NTC] [VID] v4l2_device_open: Using videodevice /dev/video0 and input -1
> Feb 07 23:39:39 ***removed***motion[11319]: [1:ml1] [NTC] [VID] v4l2_device_capability: - VIDEO_CAPTURE
> Feb 07 23:39:39 ***removed***motion[11319]: [1:ml1] [NTC] [VID] v4l2_device_capability: - STREAMING
> Feb 07 23:39:39 ***removed***motion[11319]: [1:ml1] [NTC] [VID] v4l2_input_select: Name = "Camera 1"- CAMERA
> Feb 07 23:39:39 ***removed***motion[11319]: [1:ml1] [NTC] [VID] v4l2_norm_select: Device does not support specifying PAL/NTSC norm
> Feb 07 23:39:39 ***removed***motion[11319]: [1:ml1] [NTC] [VID] v4l2_pixfmt_select: Configuration palette index 17 (YU12) for 640x480 doesn't work.
> Feb 07 23:39:39 ***removed***motion[11319]: [1:ml1] [NTC] [VID] v4l2_pixfmt_select: Supported palettes:
> Feb 07 23:39:39 ***removed***motion[11319]: [1:ml1] [NTC] [VID] v4l2_pixfmt_select: (0) YUYV (YUYV 4:2:2)
> Feb 07 23:39:39 ***removed***motion[11319]: [1:ml1] [NTC] [VID] v4l2_pixfmt_select: (1) MJPG (Motion-JPEG)
> Feb 07 23:39:39 ***removed***motion[11319]: [1:ml1] [NTC] [VID] v4l2_pixfmt_set: Testing palette MJPG (640x480)
> Feb 07 23:39:39 ***removed***motion[11319]: [1:ml1] [NTC] [VID] v4l2_pixfmt_set: Using palette MJPG (640x480)
> Feb 07 23:39:39 ***removed***motion[11319]: [1:ml1] [NTC] [VID] v4l2_pixfmt_select: Selected palette MJPG
> Feb 07 23:39:39 ***removed***motion[11319]: [1:ml1] [INF] [VID] v4l2_ctrls_list: ---------Controls---------
> Feb 07 23:39:39 ***removed***motion[11319]: [1:ml1] [INF] [VID] v4l2_ctrls_list:   V4L2 ID   Name and Range
> Feb 07 23:39:39 ***removed***motion[11319]: [1:ml1] [INF] [VID] v4l2_ctrls_list: ID09963776 Brightness, 0 to 255
> Feb 07 23:39:39 ***removed***motion[11319]: [1:ml1] [INF] [VID] v4l2_ctrls_list: ID09963777 Contrast, 0 to 255
> Feb 07 23:39:39 ***removed***motion[11319]: [1:ml1] [INF] [VID] v4l2_ctrls_list: ID09963778 Saturation, 0 to 255
> Feb 07 23:39:39 ***removed***motion[11319]: [1:ml1] [INF] [VID] v4l2_ctrls_list: ID09963788 White Balance Temperature, Auto, 0 to 1
> Feb 07 23:39:39 ***removed***motion[11319]: [1:ml1] [INF] [VID] v4l2_ctrls_list: ID09963795 Gain, 0 to 255
> Feb 07 23:39:39 ***removed***motion[11319]: [1:ml1] [INF] [VID] v4l2_ctrls_list: ID09963800 Power Line Frequency, 0 to 2
> Feb 07 23:39:39 ***removed***motion[11319]: [1:ml1] [INF] [VID] v4l2_ctrls_list:   menu item: Value 0 Disabled
> Feb 07 23:39:39 ***removed***motion[11319]: [1:ml1] [INF] [VID] v4l2_ctrls_list:   menu item: Value 1 50 Hz
> Feb 07 23:39:39 ***removed***motion[11319]: [1:ml1] [INF] [VID] v4l2_ctrls_list:   menu item: Value 2 60 Hz
> Feb 07 23:39:39 ***removed***motion[11319]: [1:ml1] [INF] [VID] v4l2_ctrls_list: ID09963802 White Balance Temperature, 0 to 10000
> Feb 07 23:39:39 ***removed***motion[11319]: [1:ml1] [INF] [VID] v4l2_ctrls_list: ID09963803 Sharpness, 0 to 255
> Feb 07 23:39:39 ***removed***motion[11319]: [1:ml1] [INF] [VID] v4l2_ctrls_list: ID09963804 Backlight Compensation, 0 to 1
> Feb 07 23:39:39 ***removed***motion[11319]: [1:ml1] [INF] [VID] v4l2_ctrls_list: ID10094849 Exposure, Auto, 0 to 3
> Feb 07 23:39:39 ***removed***motion[11319]: [1:ml1] [INF] [VID] v4l2_ctrls_list:   menu item: Value 1 Manual Mode
> Feb 07 23:39:39 ***removed***motion[11319]: [1:ml1] [INF] [VID] v4l2_ctrls_list:   menu item: Value 3 Aperture Priority Mode
> Feb 07 23:39:39 ***removed***motion[11319]: [1:ml1] [INF] [VID] v4l2_ctrls_list: ID10094850 Exposure (Absolute), 1 to 10000
> Feb 07 23:39:39 ***removed***motion[11319]: [1:ml1] [INF] [VID] v4l2_ctrls_list: ID10094851 Exposure, Auto Priority, 0 to 1
> Feb 07 23:39:39 ***removed***motion[11319]: [1:ml1] [INF] [VID] v4l2_ctrls_list: --------------------------
> Feb 07 23:39:39 ***removed***motion[11319]: [1:ml1] [NTC] [ALL] image_ring_resize: Resizing pre_capture buffer to 1 items
> Feb 07 23:39:41 ***removed***motion[11319]: [1:ml1] [NTC] [ALL] image_ring_resize: Resizing pre_capture buffer to 4 items

Next, the following logs capture the event starting (log level 9):

> Feb 07 23:37:54 ***removed*** motion[3039]: [1:ml1] [INF] [EVT] event_new_video: Source FPS 15
> Feb 07 23:37:54 ***removed*** motion[3039]: [1:ml1] [NTC] [ALL] motion_detected: Motion detected - starting event 9
> Feb 07 23:37:54 ***removed*** motion[3039]: [1:ml1] [DBG] [EVT] exec_command: Executing external command '/home/***removed***/test.sh'
> Feb 07 23:37:54 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: 1) vid_source->pframe 1
> Feb 07 23:37:54 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: 2) vid_source->pframe 1
> Feb 07 23:37:54 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: 3) vid_source->pframe 2 vid_source->buf.index 2
> Feb 07 23:37:54 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: the_buffer index 2 Address (2384c000)
> Feb 07 23:37:54 ***removed*** motion[3039]: [1:ml1] [DBG] [EVT] exec_command: Executing external command '/home/***removed***/test.sh'
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: 1) vid_source->pframe 2
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: 2) vid_source->pframe 2
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: 3) vid_source->pframe 3 vid_source->buf.index 3
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: the_buffer index 3 Address (237f8000)
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [EVT] exec_command: Executing external command '/home/***removed***/test.sh'
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: 1) vid_source->pframe 3
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: 2) vid_source->pframe 3
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: 3) vid_source->pframe 0 vid_source->buf.index 0
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: the_buffer index 0 Address (238f4000)
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [EVT] exec_command: Executing external command '/home/***removed***/test.sh'
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: 1) vid_source->pframe 0
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: 2) vid_source->pframe 0
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: 3) vid_source->pframe 1 vid_source->buf.index 1
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: the_buffer index 1 Address (238a0000)
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [EVT] exec_command: Executing external command '/home/***removed***/test.sh'
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: 1) vid_source->pframe 1
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: 2) vid_source->pframe 1
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: 3) vid_source->pframe 2 vid_source->buf.index 2
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: the_buffer index 2 Address (2384c000)
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [EVT] exec_command: Executing external command '/home/***removed***/test.sh'
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: 1) vid_source->pframe 2
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: 2) vid_source->pframe 2
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: 3) vid_source->pframe 3 vid_source->buf.index 3
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: the_buffer index 3 Address (237f8000)
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [EVT] exec_command: Executing external command '/home/***removed***/test.sh'
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: 1) vid_source->pframe 3
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: 2) vid_source->pframe 3
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: 3) vid_source->pframe 0 vid_source->buf.index 0
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: the_buffer index 0 Address (238f4000)
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [EVT] exec_command: Executing external command '/home/***removed***/test.sh'
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: 1) vid_source->pframe 0
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: 2) vid_source->pframe 0
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: 3) vid_source->pframe 1 vid_source->buf.index 1
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: the_buffer index 1 Address (238a0000)
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [EVT] exec_command: Executing external command '/home/***removed***/test.sh'
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: 1) vid_source->pframe 1
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: 2) vid_source->pframe 1
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: 3) vid_source->pframe 2 vid_source->buf.index 2
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: the_buffer index 2 Address (2384c000)
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [EVT] exec_command: Executing external command '/home/***removed***/test.sh'
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: 1) vid_source->pframe 2
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: 2) vid_source->pframe 2
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: 3) vid_source->pframe 3 vid_source->buf.index 3
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: the_buffer index 3 Address (237f8000)
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [EVT] exec_command: Executing external command '/home/***removed***/test.sh'
> Feb 07 23:37:55 ***removed*** motion[3039]: [1:ml1] [DBG] [VID] v4l2_capture: 1) vid_source->pframe 3

The logs state clearly that the command is attempting to be executed, but this does not appear to successful.

**Attempts at a solution**

I read in the [configuration guide](https://motion-project.github.io/motion_config.html), the following excerpt:

> This can be any type of program or script. Remember to set the execution bit in the > ACL and if it is a script type program such as perl or bash also remember the 
> shebang line (e.g. #!/user/bin/perl) as the first line of the script. 

I have the #! line in my script (see above), and I used the setfacl to change ACL permissions on my file test.sh (also in the home folder of my user): `sudo  setfacl -m g:motion:rwx test.sh`.

The output of `sudo getfactl test.sh` is:

> \# file: test.sh
> \# owner: motion
> \# group: motion
> user::rwx
> group::rwx
> group:motion:rwx
> mask::rwx
> other::---

I modified the permissions for my script to be owned by the motion group and user:
> -rwxrwx---+  1 motion  motion         64 Feb  7 22:38  test.sh

and, as the above line suggests, the user and owner can read, write and execute the shell script.

Just to be sure, I also gave ownership of the file to be copied to the Motion group and user:

> -rw-rw-r--+ 1 motion motion 0 Feb  3 18:57 test_file1

**Configuration and system files**

This may not be relevant, but here is the systemd service file I am using:

> [Unit]
> Description=Motion - monitor live video, trigger responses, record video/stills.
> Documentation=man:motion(1)
> After=local-fs.target network.target
> 
> [Service]
> User=motion
> EnvironmentFile=-/etc/default/motion
> \# The the sleep on the following line is needed with systemd version
> \# 232, otherwise the error message that we echo doesn't hit 'systemctl
> \# status'.
> ExecStart=/bin/sh -c '([ -r "/etc/default/motion" ] && [ "x$start_motion_daemon" != "xyes" ]) && echo "Not starting motion because start_motion_daemon=$start_motion_daemon - check /etc/default/motion" && sleep 1 && exit 78 ; exec /usr/bin/motion -n'
> Type=simple
> \# Set StandardError=journal to use journald to log messages from motion.
> \# See also the "log_file" config file option in motion(1) and
> \# systemd.service(5).
> StandardError=null
> ExecReload=/bin/kill -HUP $MAINPID
> Restart=on-failure
> RestartSec=5
> \# Don't restart if unconfigured / misconfigured  e.g. daemon disabled
> \# in defaults file.  See also /usr/include/sysexits.h or sysexits(3)
> RestartPreventExitStatus=78
> \# To tune restart behaviour, see systemd.unit(5) and use
> \# "systemctl edit motion" to change the following settings:
> \#StartLimitBurst=
> \#StartLimitIntervalSec=
> \#StartLimitAction=
> \#FailureAction=
> \# The following can be used to increase the security of the
> \# installation, by mitigating risk from attacks on motion and the
> \# binaries, libraries and scripts which it relies on.  They are disabled
> \# by default in case they break existing installations, e.g. those which
> \# call site-local scripts which would inherit the same restrictions.
> \#
> \# See systemd.exec(5) and
> \# http://0pointer.net/public/systemd-nluug-2014.pdf for more details
> \# of these and other related options.
> \#
> \# Use "systemctl edit motion" to change these settings.
> \#PrivateTmp=true
> \#NoNewPrivileges=yes
> \#PrivateNetwork=yes
> \#ProtectHome=yes
> \#DeviceAllow=/dev/video0
> \#MountFlags=slave
> \#SystemCallFilter=
> 
> [Install]
> WantedBy=multi-user.target

Here's my full configuration file as well:

> \# Rename this distribution example file to motion.conf
> \#
> \# This config file was generated by motion 4.2.2
> \# Documentation:  /usr/share/doc/motion/motion_guide.html
> \#
> \# This file contains only the basic configuration options to get a
> \# system working.  There are many more options available.  Please
> \# consult the documentation for the complete list of all options.
> \#
> 
> \############################################################
> \# System control configuration parameters
> \############################################################
> 
> \# Start in daemon (background) mode and release terminal.
> daemon on
> 
> \# Start in Setup-Mode, daemon disabled.
> setup_mode off
> 
> \# File to store the process ID.
> ; pid_file value
> 
> \# File to write logs messages into.  If not defined stderr and syslog is used.
> ; log_file value /var/log/motion/motion.log
> 
> \# Level of log messages [1..9] (EMG, ALR, CRT, ERR, WRN, NTC, INF, DBG, ALL).
> log_level 9
> 
> \# Target directory for pictures, snapshots and movies
> target_dir /var/lib/motion/captures
> 
> \# Video device (e.g. /dev/video0) to be used for capturing.
> videodevice /dev/video0
> 
> \# Parameters to control video device.  See motion_guide.html
> ; vid_control_params value
> 
> \# The full URL of the network camera stream.
> ; netcam_url value
> 
> \# Name of mmal camera (e.g. vc.ril.camera for pi camera).
> ; mmalcam_name value
> 
> \# Camera control parameters (see raspivid/raspistill tool documentation)
> ; mmalcam_control_params value
> 
> \############################################################
> \# Image Processing configuration parameters
> \############################################################
> 
> \# Image width in pixels.
> width 640
> 
> \# Image height in pixels.
> height 480
> 
> \# Maximum number of frames to be captured per second.
> framerate 15
> 
> \# Text to be overlayed in the lower left corner of images
> text_left CAMERA1
> 
> \# Text to be overlayed in the lower right corner of images.
> text_right %Y-%m-%d\n%T-%q
> 
> \############################################################
> \# Motion detection configuration parameters
> \############################################################
> 
> \# Always save pictures and movies even if there was no motion.
> emulate_motion off
> 
> \# Threshold for number of changed pixels that triggers motion.
> threshold 1500
> 
> \# Noise threshold for the motion detection.
> ; noise_level 32
> 
> \# Despeckle the image using (E/e)rode or (D/d)ilate or (l)abel.
> despeckle_filter EedDl
> 
> \# Number of images that must contain motion to trigger an event.
> minimum_motion_frames 1
> 
> \# Gap in seconds of no motion detected that triggers the end of an event.
> event_gap 60
> 
> \# The number of pre-captured (buffered) pictures from before motion.
> pre_capture 3
> 
> \# Number of frames to capture after motion is no longer detected.
> post_capture 0
> 
> \############################################################
> \# Script execution configuration parameters
> \############################################################
> 
> \# Command to be executed when an event starts.
> on_motion_detected /home/***removed***/test.sh
> 
> \# Command to be executed when an event ends.
> ; on_event_end value
> 
> \# Command to be executed when a movie file is closed.
> ; on_movie_end value
> 
> \############################################################
> \# Picture output configuration parameters
> \############################################################
> 
> \# Output pictures when motion is detected
> picture_output off
> 
> \# File name(without extension) for pictures relative to target directory
> picture_filename %Y%m%d%H%M%S-%q
> 
> \############################################################
> \# Movie output configuration parameters
> \############################################################
> 
> \# Create movies of motion events.
> movie_output off
> 
> \# Maximum length of movie in seconds.
> movie_max_time 60
> 
> \# The encoding quality of the movie. (0=use bitrate. 1=worst quality, 100=best)
> movie_quality 45
> 
> \# Container/Codec to used for the movie. See motion_guide.html
> movie_codec mkv
> 
> \# File name(without extension) for movies relative to target directory
> movie_filename %t-%v-%Y%m%d%H%M%S
> 
> \############################################################
> \# Webcontrol configuration parameters
> \############################################################
> 
> \# Port number used for the webcontrol.
> webcontrol_port 8080
> 
> \# Restrict webcontrol connections to the localhost.
> webcontrol_localhost on
> 
> \# Type of configuration options to allow via the webcontrol.
> webcontrol_parms 0
> 
> \############################################################
> \# Live stream configuration parameters
> \############################################################
> 
> \# The port number for the live stream.
> stream_port 8081
> 
> \# Restrict stream connections to the localhost.
> stream_localhost on
> 
> \##############################################################
> \# Camera config files - One for each camera.
> \##############################################################
> ; camera /etc/motion/camera1.conf
> ; camera /etc/motion/camera2.conf
> ; camera /etc/motion/camera3.conf
> ; camera /etc/motion/camera4.conf
> 
> \##############################################################
> \# Directory to read '.conf' files for cameras.
> \##############################################################
> ; camera_dir /etc/motion/conf.d

Is there a permissions directive from the manual I missed?  Thank you for your time in helping me solve this issue.

**Comment by tosiara** [comment]
You need to check whether your script is not being executed, or your copy command inside is not doing what you are expecting

1. Try to run motion standalone, not from daemon, like `motion -c myconf.conf`
2. Try `strace motion -c myconf.conf` - do you see your script really executing?
3. Try to put `touch /tmp/evidence.txt` in you script - does it create a file in `/tmp`?

**Comment by CaptainMayotoast** [comment]
@tosiara thank you for your reply!  

I tried all three of your suggested steps (and had to add the motion user to the video group), my script copied the file as I expected, and created evidence.txt in /tmp.

I would like the Motion service to exhibit this same behavior, however.  What are your thoughts on what could be the difference between the non-daemon mode and the service?  In both cases, /usr/bin/motion and /etc/motion/motion.conf are used.

I am generally unfamiliar with how systemd handles users and permissions - correct me if I am wrong, but when I ran Motion as you suggested, wasn't I running Motion as my current user?  Wouldn't this then suggest that there are permissions issues when systemd runs Motion with the Motion user?

Thank you for your time, and I look forward to your reply.
